service/src/main/java/uk/gov/mca/beacons/api/registration/application/RegistrationReadOnlyService.java

```diff
@@ -79,7 +81,66 @@ public Registration getByBeaconId(BeaconId beaconId) {
       .build();
   }
 
-  public List<Note> getNonSyetemNotesByBeaconId(BeaconId beaconId) {
+  public Registration getRegistrationFromBeaconBackupItem(
+    BeaconBackupItem beaconBackupItem
+  ) {
+    Registration registration = Registration.builder().build();
+    registration =
+      hydrateBeaconFromBeaconBackupItem(registration, beaconBackupItem);
+    BeaconId beaconId = registration.getBeacon().getId();
+
+    BeaconOwner beaconOwner = beaconOwnerRepository
+      .findBeaconOwnerByBeaconId(beaconId)
+      .orElse(null);
+    List<BeaconUse> beaconUses = beaconUseRepository.findBeaconUsesByBeaconId(
+      beaconId
+    );
+    List<EmergencyContact> emergencyContacts = emergencyContactRepository.findEmergencyContactsByBeaconId(
+      beaconId
+    );
+
+    registration.setBeaconOwner(beaconOwner);
+    registration.setBeaconUses(beaconUses);
+    registration.setEmergencyContacts(emergencyContacts);
+
+    return registration;
+  }
+
+  private Registration hydrateBeaconFromBeaconBackupItem(
+    Registration registration,
+    BeaconBackupItem beaconBackupItem
+  ) {
+    Beacon beacon = new Beacon();
+    beacon.setId(new BeaconId(beaconBackupItem.getId()));
```

Comment:
Would it also make more sense to have this as a builder method?

Nothing particularly wrong with how it is at the moment, thinking more from a consistency pov

Reply:
Originally I did it as a builder method yeah adding the @Builder. However, applying involves changing older parts of the codebase where Zack and Olly used new Beacon(). Is it better for us to be consistent with the old style or with the way we've used so far in the export code?